### PR TITLE
[GitHub] Issue and pull request detail and check state

### LINKED ISSUE #1011: Open/Closed State badge for github issues.
I just forked the repo to add open/closed issue state badges, i'll send a PR when its complete... 

this will be useful for our Readme file's known issue section.

https://github.com/evancohen/smart-mirror/blob/master/README.md

### PR DESCRIPTION
**Pull request by paulmelnikow**
This adds badges for Github issues and pull requests. You can display the state, title, username, number of comments, age, time since last update, and state of checks.

Blocked on #1112 (see FIXMEs).

Provides an endpoint the Shields CI can use to fetch PR titles for #979 and resolves #1011.

### CODE REVIEW

File: service-tests/github.js [hunk 1]
```diff
@@ -233,3 +233,58 @@ t.create('hit counter for nonexistent repo')
...
+t.create('github issue age')
+  .get('/issues/detail/age/badges/shields/979.json')
+  .expectJSONTypes(Joi.object().keys({ name: 'created', value: validDateString }));
+
+t.create('github issue update')
+  .get('/issues/detail/last-update/badges/shields/979.json')
+  .expectJSONTypes(Joi.object().keys({ name: 'updated', value: validDateString }));
+
+t.create('github pull request check state')
+  .get('/pulls/checks/s/badges/shields/1110.json')
```
Comment (paulmelnikow) [inline service-tests/github.js]: We could make this `/status/s/pulls/badges/shields/1110.json` which might be nicer for adding check state of branches + commits, as in #774.

File: server.js [hunk 2]
```diff
@@ -3496,6 +3497,134 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// GitHub issue detail integration.
+camp.route(/^\/github\/(?:issues|pulls)\/detail\/(s|title|u|label|comments|age|last-update)\/([^\/]+)\/([^\/]+)\/(\d+)\.(svg|png|gif|jpg|json)$/,
+cache((queryParams, match, sendBadge, request) => {
+  const stateColor = s => ({ open: '2cbe4e', closed: 'cb2431', merged: '6f42c1' }[s]);
```
Comment (Daniel15) [inline server.js]: This is going to re-define the function ever time this route is called, which is not really necessary as it's entirely static. Instead, pull this function out and make it a regular function in this file (or in one of the utility modules).

Comment (paulmelnikow) [reply server.js]: Updated in 448e88c.

File: service-tests/github.js (hunk 1 above)
Comment (paulmelnikow) [reply service-tests/github.js]: Updated: c727f2f.

**Comment by paulmelnikow** [comment]
Service tests passing locally with [this patch](https://github.com/paulmelnikow/shields/tree/gh-token) that works around a bug in our logic for token rate limit handling.

**Pull request merged**